charliecloud: init at 0.9.8

bzizou opened the pull request:
###### Motivation for this change
charliecloud is used into our supercomputing center

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

bzizou committed:
charliecloud: init at 0.9.8

markuskowa commented on pkgs/applications/virtualization/charliecloud/default.nix:
```diff
@@ -0,0 +1,39 @@
+
+  preConfigure = ''
+    substituteInPlace Makefile --replace '/bin/bash' '${bash}/bin/bash'
+    for file in test/* ; do
```
Would `patchShebangs test/` not be shorter and achieve the same thing?

markuskowa commented on pkgs/applications/virtualization/charliecloud/default.nix:
```diff
@@ -0,0 +1,39 @@
+  buildInputs = [ python ];
+
+  preConfigure = ''
+    substituteInPlace Makefile --replace '/bin/bash' '${bash}/bin/bash'
```
```suggestion
    substituteInPlace Makefile --replace '/bin/bash' '${stdenv.shell}'
```

markuskowa commented on pkgs/applications/virtualization/charliecloud/default.nix:
```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchFromGitHub, python, bash }:
```
```suggestion
{ stdenv, fetchFromGitHub, python }:
```

markuskowa commented on pkgs/applications/virtualization/charliecloud/default.nix:
```diff
@@ -0,0 +1,39 @@
+  meta = {                                                                                   
+    description = "User-defined software stacks (UDSS) for high-performance computing (HPC) centers";
+    longDescription = ''
+      Charliecloud uses Linux user namespaces to run containers with no privileged operations or daemons and minimal configuration changes on center resources. This simple approach avoids most security risks while maintaining access to the performance and functionality already on offer.''; 
```
Please insert some line breaks here.

markuskowa commented:
@GrahamcOfBorg build charliecloud

markuskowa commented:
@bzizou can you please setup your editor such that it removes trailing white spaces at the end of the line.
The folder `lib/charliecloud` contains two sub folders, `examples` and `tests`, which would fit better in `share/charliecloud`.

bzizou commented:
Thank you, I added this into my vimrc :-)
```
highlight ExtraWhitespace ctermbg=red guibg=red
match ExtraWhitespace /\s\+$/
```
And I moved the sub folders `examples` and `test` into `share/charliecloud`

markuskowa commented:
Looks good, thank you!
